### PR TITLE
[BugFix] support two level nested array type in hudi mor table

### LINKED ISSUE #19593: Hudi MOR table does not support two level nested array
### Steps to reproduce the behavior (Required)

```
create external catalog hudi_catalog_8814331c_c172_11ed_96c2_00163e0e550b
PROPERTIES("type"="hudi",
"hive.metastore.uris"="thrift://172.26.195.10:9083");

use hudi_catalog_8814331c_c172_11ed_96c2_00163e0e550b.hudi_db_struct_map;

select a,b,c,d,e from hudi_mor_jniscan_parquet1;
```

### Expected behavior (Required)

select out data

### Real behavior (Required)

BE crashes, stacktrace is like

```
*** Aborted at 1678922088 (unix time) try "date -d @1678922088" if you are using GNU date ***
PC: @         0x18fc1c64 std::__cxx11::basic_string<>::end()
*** SIGSEGV (@0x0) received by PID 3652925 (TID 0x7f897895b700) from PID 0; stack trace: ***
    @         0x16e5b7a2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f8b283a5e9b os::Linux::chained_handler()
    @     0x7f8b283aa90c JVM_handle_linux_signal
    @     0x7f8b2839d858 signalHandler()
    @     0x7f8b27177630 (unknown)
    @         0x18fc1c64 std::__cxx11::basic_string<>::end()
    @          0xd454952 boost::range_detail::range_end<>()
    @          0xd452e67 boost::range_adl_barrier::end<>()
    @          0xebec887 boost::algorithm::detail::transform_range_copy<>()
    @          0xebea299 boost::algorithm::to_lower_copy<>()
    @         0x12dbc6d8 _ZN9starrocks7parquet12ColumnReader6createERKNS0_19ColumnReaderOptionsEPKNS0_12ParquetFieldERKNS_14TypeDescriptorEPSt10unique_ptrIS1_St14default_deleteIS1_EE.localalias
    @         0x12dbb4ed _ZN9starrocks7parquet12ColumnReader6createERKNS0_19ColumnReaderOptionsEPKNS0_12ParquetFieldERKNS_14TypeDescriptorEPSt10unique_ptrIS1_St14default_deleteIS1_EE.localalias
    @         0x130b9496 starrocks::parquet::GroupReader::_create_column_reader()
    @         0x130b8fd8 starrocks::parquet::GroupReader::_init_column_readers()
    @         0x130b630f starrocks::parquet::GroupReader::init()
    @         0x130d6e5b starrocks::parquet::FileReader::_init_group_readers()
    @         0x130cd1a6 starrocks::parquet::FileReader::init()
    @          0xec171cc starrocks::HdfsParquetScanner::do_open()
    @          0xebe04f9 starrocks::HdfsScanner::open()
    @          0xda300a4 starrocks::connector::HiveDataSource::_init_scanner()
    @          0xda23144 starrocks::connector::HiveDataSource::open()
    @          0xf4de4d7 starrocks::pipeline::ConnectorChunkSource::_open_data_source()
    @          0xf4de93b starrocks::pipeline::ConnectorChunkSource::_read_chunk()
    @          0xf4337a6 starrocks::pipeline::ChunkSource::buffer_next_batch_chunks_blocking()
    @          0xf470fb4 _ZZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS_12RuntimeStateEiENKUlvE_clEv
    @          0xf47696a _ZSt13__invoke_implIvRZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS0_12RuntimeStateEiEUlvE_JEET_St14__invoke_otherOT0_DpOT1_
    @          0xf476800 _ZSt10__invoke_rIvRZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS0_12RuntimeStateEiEUlvE_JEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EES8_E4typeEOS9_DpOSA_
    @          0xf47633a _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS1_12RuntimeStateEiEUlvE_E9_M_invokeERKSt9_Any_data
    @          0xd59a3fa std::function<>::operator()()
    @          0xf98c239 starrocks::workgroup::ScanExecutor::worker_thread()
    @          0xf98b9ae _ZZN9starrocks9workgroup12ScanExecutor10initializeEiENKUlvE_clEv
    @          0xf98d53e _ZSt13__invoke_implIvRZN9starrocks9workgroup12ScanExecutor10initializeEiEUlvE_JEET_St14__invoke_otherOT0_DpOT1_
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19593

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by zombee0 — DISMISSED** [review]

**Comment by dirtysalt** [comment]
run starrocks_be_unittest

**Review by murphyatwork — DISMISSED** [review]

**Comment by dirtysalt** [comment]
run starrocks_admit_test

File: be/src/formats/parquet/schema.cpp [hunk 1]
```diff
@@ -125,9 +125,9 @@ Status SchemaDescriptor::list_to_field(const std::vector<tparquet::SchemaElement
     //     }
     //   }
     auto& level1_schema = t_schemas[pos];
-    if (level1_schema.repetition_type == tparquet::FieldRepetitionType::REPEATED) {
-        return Status::InvalidArgument("LIST-annotated group must be not repeated");
-    }
+    // if (level1_schema.repetition_type == tparquet::FieldRepetitionType::REPEATED) {
```
Comment (chaoyli) [inline be/src/formats/parquet/schema.cpp:128]: The annotation codes paths can be removed.

Comment (Youngwb) [inline be/src/formats/parquet/schema.cpp:128]: remove this？

File: be/src/formats/parquet/schema.cpp [hunk 2]
```diff
@@ -172,6 +172,8 @@ Status SchemaDescriptor::list_to_field(const std::vector<tparquet::SchemaElement
             // }
             if (is_map(level2_schema)) {
                 RETURN_IF_ERROR(map_to_field(t_schemas, pos + 1, cur_level_info, element, next_pos));
+            } else if (is_list(level2_schema)) {
```
Comment (Youngwb) [inline be/src/formats/parquet/schema.cpp:175]: If level2_schema could be list, the comment above looks like not accurate?

Comment (dirtysalt) [reply be/src/formats/parquet/schema.cpp:175]: I 'll update this. I don't change comment.

**Review by Youngwb — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**